services/worker-manager/src/provisioner.js

```diff
@@ -158,10 +163,18 @@ class Provisioner {
           }
           poolsByProvider.get(providerId).add(workerPoolId);
 
-          const providerByPool = providersByPool.get(workerPoolId) || {providers: new Set(), capacity: 0};
+          const providerByPool = providersByPool.get(workerPoolId) || {
+            providers: new Set(),
```

Comment:
how come this `Set` is empty?

Reply:
It's the default value if there's no workers found for that workerPoolId.